app/src/main/java/com/sschuraytz/charcoaldrawing/VoiceCommands.java

```diff
@@ -0,0 +1,207 @@
+
+            @Override
+            public void onEndOfSpeech() {
+                voiceListener.updateFABUI();
+            }
+
+            @Override
+            public void onError(int error) {
+                voiceListener.updateFABUI();
+            }
+
+            //Is there a way to make it more flexible so if user doesn't say exact word it still works?
+            @Override
+            public void onResults(Bundle results) {
+                ArrayList<String> data = results.getStringArrayList(SpeechRecognizer.RESULTS_RECOGNITION);
+                if (data != null) {
+                    String result = data.get(0);
+                    switch (result) {
+                        case "charcoal":
+                        case "draw":
+                            Toast.makeText(context, result, Toast.LENGTH_SHORT).show();
+                            voiceListener.charcoalCommand();
+                            break;
+                        case "eraser":
+                        case "erase":
+                            Toast.makeText(context, result, Toast.LENGTH_SHORT).show();
+                            voiceListener.eraserCommand();
+                            break;
+                        case "undo":
+                            Toast.makeText(context, result, Toast.LENGTH_SHORT).show();
+                            voiceListener.undoCommand();
+                            break;
+                        case "redo":
+                            Toast.makeText(context, result, Toast.LENGTH_SHORT).show();
+                            voiceListener.redoCommand();
+                            break;
+                        case "new":
+                        case "new canvas":
+                            Toast.makeText(context, result, Toast.LENGTH_SHORT).show();
+                            voiceListener.createNewCanvasCommand();
+                            break;
+                        default:
+                            // slider/radius
+                            if (isNumber(result)) {
+                                if (Integer.parseInt(result) < 0 || Integer.parseInt(result) > 100) {
+                                    Toast.makeText(context, "Width must be between 0 and 100", Toast.LENGTH_SHORT).show();
+                                }
+                                else {
+                                    Toast.makeText(context, result, Toast.LENGTH_SHORT).show();
+                                    voiceListener.updateDrawingThickness(Integer.parseInt(result));
+                                }
+                            }
+                            else {
+                                Toast.makeText(context, "no such command", Toast.LENGTH_SHORT).show();
+                            }
+                    }
+                    //smudge
+                    //save
+                    //help
+                }
+            }
+
+            public boolean isNumber(String str) {
```

Comment:
I wouldn't write this yourself.
Here are some alternatives.
https://www.baeldung.com/java-check-string-number